src/generator/mod.rs

```diff
@@ -31,14 +31,19 @@ impl NamedTxRequest {
     pub fn with_name(name: &str, tx: TransactionRequest) -> Self {
         Self {
             name: Some(name.to_string()),
+            kind: None,
             tx,
         }
     }
+
+    pub fn set_kind(&mut self, kind: String) {
+        self.kind = Some(kind);
+    }
```

Comment:
I think this all should just be swapped out for a `new` constructor.

```rs
impl NamedTxRequest {
  pub fn new(tx: TransactionRequest, name: Option<&str>, kind: Option<&str>) {...}
}
```

This is more idiomatic, and we don't have any reason to iteratively build the struct in our code in the first place.